src/history/config.go

```diff
@@ -164,8 +166,16 @@ func NewService() *Service {
 	}
 }
 
-var history = make(map[string]*Service)
-var mux sync.Mutex
+var (
+	history = make(map[string]*Service)
+	mux     sync.Mutex
+	tagHash = sha256.Sum256
+)
+
+func makeTag(repoURL string, state State) string {
+	data := []byte(fmt.Sprintf("%v%v", state.Timestamp.Format(TimeFormatString), repoURL))
```

Comment:
Ah, the timestamp changes on every healthcheck. See history.go line 94.
Can you not depend on the timestamp or add two timestamps to the state struct - ActionUpdateTs and HealthUpdateTs ?

Reply:
An easy way would be to just use the <status,subdomain,access,server> tuple instead of the timestamp. A lot of things will need to be changed to split the timestamp. You can ignore for the time-being.

Reply:
Yeah, I'll do that. It actually might be acceptable to set the state based on this always. Since any function assumes only the previous state (and not that any change didn't happen since it retrieved the tag), it shouldn't matter if anyone else came and changed it a couple of times so it resulted in the same state again.